internal/user/user_profile.go

```diff
@@ -1,38 +1,42 @@
 package user
 
 import (
+	"math/rand"
+
+	"github.com/kopia/kopia/internal/crypto"
 	"github.com/kopia/kopia/repo/manifest"
 )
 
 // Profile describes information about a single user.
 type Profile struct {
 	ManifestID manifest.ID `json:"-"`
 
-	Username            string `json:"username"`
-	PasswordHashVersion int    `json:"passwordHashVersion"` // indicates how password is hashed
-	PasswordHash        []byte `json:"passwordHash"`
+	Username               string `json:"username"`
+	PasswordHashVersion    int    `json:"passwordHashVersion,omitempty"` // indicates how password is hashed, deprecated in favor of KeyDerivationAlgorithm
+	KeyDerivationAlgorithm string `json:"keyDerivationAlgorithm,omitempty"`
```

Comment:
@julio-lopez @jkowalski could you please take a quick look if deprecating a field in favor of a better type and name in user profile struct is ok here ?

Reply:
ADR:
- Reuse `PasswordHashVersion` and introduce a new version (`2`) for `pbkdf2-....-`
- Remove `KeyDerivationAlgorithm`

And then have a mapping from
- the string algo type to the version int to translate from the CLI to the struct
- the `PasswordHashVersion` int to the corresponding string for the algo, so its implementation can be looked up in the registered derivation implementations.

Reply:
Addressed in #3816